tests/Services/SearchServiceTest.php

```diff
@@ -219,6 +219,47 @@ public function redirectToProductPageOnDoSearchProvider(): array
                 'redirectUrl' => '/test-product/a-11',
                 'attributes' => []
             ],
+            'One product found on second page' => [
```

Comment:
It's not clear as this case is in `redirectToProductPageOnDoSearchProvider`. Maybe call it 'One product found on second page should not redirect to product detail' or similar.